x-pack/plugins/fleet/server/services/agent_policies/full_agent_policy.ts

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObjectsClientContract } from 'kibana/server';
+import { safeLoad } from 'js-yaml';
+
+import type {
+  FullAgentPolicy,
+  PackagePolicy,
+  Settings,
+  Output,
+  FullAgentPolicyOutput,
+} from '../../types';
+import { agentPolicyService } from '../agent_policy';
+import { outputService } from '../output';
+import {
+  storedPackagePoliciesToAgentPermissions,
+  DEFAULT_PERMISSIONS,
+} from '../package_policies_to_agent_permissions';
+import { storedPackagePoliciesToAgentInputs, dataTypes, outputType } from '../../../common';
+import type { FullAgentPolicyOutputPermissions } from '../../../common';
+import { getSettings } from '../settings';
+
+const MONITORING_DATASETS = [
+  'elastic_agent',
+  'elastic_agent.elastic_agent',
+  'elastic_agent.apm_server',
+  'elastic_agent.filebeat',
+  'elastic_agent.fleet_server',
+  'elastic_agent.metricbeat',
+  'elastic_agent.osquerybeat',
+  'elastic_agent.packetbeat',
+  'elastic_agent.endpoint_security',
+  'elastic_agent.auditbeat',
+  'elastic_agent.heartbeat',
+];
+
+export async function getFullAgentPolicy(
```

Comment:
I may have created some work for you by merging https://github.com/elastic/kibana/pull/112397 sorry 😓